server/src/test/java/io/crate/analyze/where/EqualityExtractorTest.java

```diff
@@ -396,7 +396,7 @@ public void test_pk_extraction_breaks_after_x_iterations() {
         EqualityExtractor extractor = new EqualityExtractor(normalizer) {
             @Override
             protected int maxIterations() {
-                return 1;
+                return 100;
```

Comment:
Why are these changes here needed?

Reply:
the 100->1000 change is needed, because with 100 now we return null matches, but before we returned an incomplete list.
the 1 -> 100 is a safe number to demonstrate that even 100 iterations are not enough, this is not necessary, but thought it's nicer than 1 where one could assume anyway will never work since we join the `x = ? AND y = ?` 20 times. If you think is unnecessary I can revert this one.

Reply:
Ah I see. Probably 2 short comments would be good here.

Reply:
thx, done!